modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractKotlinCodegen.java

```diff
@@ -771,7 +771,9 @@ protected boolean isReservedWord(String word) {
     @Override
     protected boolean needToImport(String type) {
         // provides extra protection against improperly trying to import language primitives and java types
-        boolean imports = !type.startsWith("kotlin.") && !type.startsWith("java.") && !defaultIncludes.contains(type) && !languageSpecificPrimitives.contains(type);
+        boolean imports = !type.startsWith("kotlin.") && !type.startsWith("java.") &&
+                !defaultIncludes.contains(type) && !languageSpecificPrimitives.contains(type) &&
+                !type.contains(".");
```

Comment:
Interesting. I was trying to think of a case where this would fall through, but can't really. Seems like any type with a dot in the name can be assumed to be fully qualified.

If people report it as a bug, we can always adjust. In that unlikely event, it may result in a regression since the change isn't covered by a unit test.